library/std/src/sys/anonymous_pipe/mod.rs

```diff
@@ -0,0 +1,103 @@
+use crate::{io, process::Stdio, sys::pipe::AnonPipe};
+
+/// Create annoymous pipe that is close-on-exec and blocking.
+#[unstable(feature = "anonymous_pipe", issue = "127154")]
+#[inline]
+pub fn pipe() -> io::Result<(PipeReader, PipeWriter)> {
+    cfg_if::cfg_if! {
+        if #[cfg(unix)] {
+            unix::pipe()
+        } else if #[cfg(windows)] {
+            windows::pipe()
+        } else {
+            panic!("Anonymous pipe is not supported on this target!")
+        }
+    }
+}
+
+/// Read end of the annoymous pipe.
+#[unstable(feature = "anonymous_pipe", issue = "127154")]
+#[derive(Debug)]
+pub struct PipeReader(AnonPipe);
+
+/// Write end of the annoymous pipe.
+#[unstable(feature = "anonymous_pipe", issue = "127154")]
+#[derive(Debug)]
+pub struct PipeWriter(AnonPipe);
+
+impl PipeReader {
+    /// Create a new [`PipeReader`] instance that shares the same underlying file description.
+    #[unstable(feature = "anonymous_pipe", issue = "127154")]
+    pub fn try_clone(&self) -> io::Result<Self> {
+        self.0.try_clone().map(Self)
+    }
+}
+
+impl PipeWriter {
+    /// Create a new [`PipeWriter`] instance that shares the same underlying file description.
+    #[unstable(feature = "anonymous_pipe", issue = "127154")]
+    pub fn try_clone(&self) -> io::Result<Self> {
+        self.0.try_clone().map(Self)
+    }
+}
+
+macro_rules! forward_io_read_traits {
+    ($name:ty) => {
+        #[unstable(feature = "anonymous_pipe", issue = "127154")]
+        impl io::Read for $name {
+            fn read(&mut self, buf: &mut [u8]) -> io::Result<usize> {
+                self.0.read(buf)
+            }
+            fn read_vectored(&mut self, bufs: &mut [io::IoSliceMut<'_>]) -> io::Result<usize> {
+                self.0.read_vectored(bufs)
+            }
+            #[inline]
+            fn is_read_vectored(&self) -> bool {
+                self.0.is_read_vectored()
+            }
+            fn read_to_end(&mut self, buf: &mut Vec<u8>) -> io::Result<usize> {
+                self.0.read_to_end(buf)
+            }
+            fn read_buf(&mut self, buf: io::BorrowedCursor<'_>) -> io::Result<()> {
+                self.0.read_buf(buf)
+            }
+        }
+    };
+}
+forward_io_read_traits!(PipeReader);
+forward_io_read_traits!(&PipeReader);
```

Comment:
Using a macro for two impls is more harmful than helpful and not consistent with what is done elsewhere in `std` anyway.

(Same for `forward_io_write_traits!` below and `impl_traits!` in other files)

Reply:
Removed.